### PR TITLE
[MM-17372] `Uncaught ReferenceError: regeneratorRuntime is not defined`

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- required a dependency update

#### Links
- Fixes [MM-17372](https://mattermost.atlassian.net/browse/MM-17372)

### CODE REVIEW

**Comment by cpoile** [comment]
@crspeller I'm doing a cargo-culting here -- I'm not sure if we need all these dependencies.

**Comment by cpoile** [comment]
The eslint errors are incorrect, at least as far as using super with async/await. The docs wrap await in a try catch for the error: https://visionmedia.github.io/superagent/#request-basics

**Comment by mickmister** [comment]
@cpoile the docs wrap with try-catch, but they also imply that you will do something with err.message etc. The code reads kinda weird, though it is nice that we aren't "ignoring" the fact that there could be an expected error here.

File: webapp/src/client/client.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ export default class Client {
         headers['X-Requested-With'] = 'XMLHttpRequest';
         headers['X-Timezone-Offset'] = new Date().getTimezoneOffset();
 
-        try {
+        try { // eslint-disable-line no-useless-catch
```
Comment (crspeller) [inline webapp/src/client/client.js]: Why not drop all the useless catches? That's what I did with other dependency updates.

File: webapp/package.json [hunk 2]
```diff
@@ -11,19 +11,23 @@
...
-    "eslint-plugin-react": "^7.10.0",
-    "webpack": "^4.16.1",
-    "webpack-cli": "^3.0.8"
+    "@babel/cli": "7.5.0",
+    "@babel/core": "^7.5.0",
+    "@babel/plugin-proposal-class-properties": "7.5.0",
+    "@babel/plugin-proposal-object-rest-spread": "^7.5.5",
+    "@babel/plugin-syntax-dynamic-import": "7.2.0",
+    "@babel/polyfill": "7.4.4",
+    "@babel/preset-env": "^7.5.0",
```
Comment (crspeller) [inline webapp/package.json]: Generally we specify exact versions.

File: webapp/package.json [hunk 3]
```diff
@@ -11,19 +11,23 @@
...
-    "babel-plugin-transform-object-rest-spread": "^6.26.0",
-    "babel-preset-env": "^1.7.0",
-    "babel-preset-react": "^6.24.1",
-    "eslint": "^5.2.0",
-    "eslint-import-resolver-webpack": "^0.10.1",
-    "eslint-plugin-import": "^2.13.0",
-    "eslint-plugin-react": "^7.10.0",
-    "webpack": "^4.16.1",
-    "webpack-cli": "^3.0.8"
+    "@babel/cli": "7.5.0",
```
Comment (crspeller) [inline webapp/package.json]: I think there are newer versions of these. Might as well go to latest? 
I usually use npm-check to find the latest versions for me: https://www.npmjs.com/package/npm-check

File: webapp/src/client/client.js (hunk 1 above)
Comment (cpoile) [reply webapp/src/client/client.js]: Good point, done.

File: webapp/package.json (hunk 3 above)
Comment (cpoile) [reply webapp/package.json]: done!

**Pull request merged**